src/sql/parts/notebook/models/notebookModel.ts

```diff
@@ -499,6 +523,7 @@ export class NotebookModel extends Disposable implements INotebookModel {
 	public async handleClosed(): Promise<void> {
 		try {
 			if (this._activeClientSession) {
+				await this._activeClientSession.ready;
```

Comment:
If session ready is ever rejected this will cause problems. If it's not ready, can we do something to still shut it down / end this? 
- At a minimum, try/catch the ready call
- consider if awaiting this will potentially take a long time, and use a timeout to just skip or hard-shutdown the session in that case (e.g. if it's longer than 10 seconds)